File: math/gcd/gcd.js
/**
 * Copyright https://github.com/microwind
 * GCD - Greatest Common Divisor
 */

function gcd(x, y) {
    while (y !== 0) {
        const temp = y;
        y = x % y;
        x = temp;
    }
    return x;
}

function main() {
    console.log("GCD(48, 18) = " + gcd(48, 18));
    console.log("GCD(56, 98) = " + gcd(56, 98));
}

main();
if (typeof module !== 'undefined') module.exports = { gcd };

File: math/gcd/gcd.java
/**
 * Copyright https://github.com/microwind
 * GCD - Greatest Common Divisor
 */

class GCD {
    public static int gcd(int x, int y) {
        while (y != 0) {
            int temp = y;
            y = x % y;
            x = temp;
        }
        return x;
    }
    
    public static void main(String[] args) {
        System.out.println("GCD(48, 18) = " + gcd(48, 18));
        System.out.println("GCD(56, 98) = " + gcd(56, 98));
    }
}
